src/wallet/fidelity.rs

```diff
@@ -412,7 +412,7 @@ impl Wallet {
         bond.cert_expiry = Some(cert_expiry);
         bond.conf_height = Some(conf_height);
 
-        self.sync()?;
+        self.sync_and_save()?;
```

Comment:
Just curious. DO we need a sync here? Does anything break if we remove this? 

Sync calls are expensive, so we should remove if they are not strictly needed.